codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/smithy/generators/CargoTomlGenerator.kt

```diff
@@ -54,6 +54,9 @@ class CargoTomlGenerator(
         if (features.isNotEmpty()) {
             cargoFeatures.add("default" to features.filter { it.default }.map { it.name })
         }
+        // add serde related features
+        cargoFeatures.add("serialize" to listOf("aws-smithy-types/serialize"))
+        cargoFeatures.add("deserialize" to listOf("aws-smithy-types/deserialize"))
```

Comment:
I'm not super familiar with the feature flag machinery here, but are `serialize`/`deserialize` going to be the names of the feature flags?

Reply:
yup!

let me prepare you an example on another crate.

Reply:
I would suggest going for `serde-deserialize`/`serde-serialize` instead of `serialize`/`deserialize` - we might want to add support for other (de)serialization crates down the line.

Reply:
If I understand it correctly, it was agreed that name of the feature gate is going to become `serialize`/`deserialize` on RFC30.

https://github.com/awslabs/smithy-rs/blob/main/design/src/rfcs/rfc0030_serialization_and_deserialization.md

```
Feature Gate for Serialization and De-serialization

Serde traits are implemented behind feature gates. Serialize is implemented behind serialize, while Deserialize is implemented behind deserialize. Users must enable the unstable feature to expose those features.

We considered giving each feature a dedicated feature gate such as unstable-serde-serialize. In this case, we will need to change the name of feature gates entirely once it leaves the unstable status which will cause users to make changes to their code base. We conclude that this brings no benefit to the users.
```

You can find the discussion here.
https://github.com/awslabs/smithy-rs/pull/1944

Furthermore, those feature gates can be enabled only when aws_sdk_unstable is enabled via cfg flag.
like this.
```bash
export RUSTFLAGS="--cfg aws-sdk-unstable"
cargo build
```

Reply:
@Velfi 
Am I correct to believe that we are going to implement the way the RFC describes including the name of the feature-gate?

@LukeMathWalker 
@Velfi was heavily involved in the discussion. I'm going to mention her to clarify the situation.

Reply:
The RFC touches on a few crates that are co-owned by the service framework team (e.g. aws-smithy-http) - I realised that was the case too late (a.k.a. after it was merged), so that's why I'm jumping in on the implementation PR 😁 

The issue I am raising is not about us removing those flags - it is about us wanting to support, for example, the [`deser`](https://github.com/mitsuhiko/deser) crate alongside `serde` as a serialization option. Including the crate name in the feature name allows for this addition without causing confusion.

Reply:
Thank you for letting me know!

Yeah, I think you make sense; I prefer to have the RFC updated just to make sure that everyone is on the same pages.

Would you kindly create a PR to update the RFC?

Reply:
by the way, here is a repository to show case how feature-gate works.
https://github.com/thomas-k-cameron/feature-gate-examples

Reply:
https://github.com/awslabs/smithy-rs/pull/2183#discussion_r1064573442

https://github.com/awslabs/smithy-rs/pull/2186

@LukeMathWalker 

I just created the PR for myself.
can you check it out?

Reply:
Oh, I now realise the confusion - when I said I wasn't familiar with the feature gate machinery I meant that I'm not familiar with how we handle that in the Kotlin code generator, not in Rust per se 😅 Sorry for having waste your time there 😅

Reply:
oh snap

no worries :D

Well, this should work but I'm not too familiar with it too. 

That is why it is in a `draft` state.